internal/datacoord/garbage_collector.go

```diff
@@ -148,24 +147,24 @@ func (gc *garbageCollector) Resume(ctx context.Context) error {
 // work contains actual looping check logic
 func (gc *garbageCollector) work() {
 	defer gc.wg.Done()
-	ticker := time.NewTicker(gc.option.checkInterval)
-	defer ticker.Stop()
-	scanTicker := time.NewTicker(gc.option.scanInterval)
-	defer scanTicker.Stop()
+
+	// TODO: fast cancel for gc when closing.
+	// Run gc tasks in parallel.
+	gc.wg.Add(1)
+	go gc.runRecycleTaskWithPauser("meta", gc.option.checkInterval, func() {
+		gc.clearEtcd()
+		gc.recycleUnusedIndexes()
+		gc.recycleUnusedSegIndexes()
+	})
+
+	gc.wg.Add(1)
+	go gc.runRecycleTaskWithPauser("file", gc.option.scanInterval, func() {
+		gc.recycleUnusedBinlogFiles()
+		gc.recycleUnusedIndexFiles()
```

Comment:
we can not move recycleUnusedIndexFiles into 7 days interval for now.

Reason is:
for data, when you clear etcd, you also do removeLogs so data on S3 is cleaned. Scan is simply trying to find orphan files and clean. So even if you are not doing any cleanup it's still fine.

for index, recycleUnusedIndexes and recycleUnusedSegIndexes only removes data on etcd. We can not wait until 7 days to cleanup index on S3. Maybe we can add similar logic to clean data on S3 also but we really need to be careful

Reply:
recycleUnusedBinlogFiles and recycleUnusedIndexFiles should be named to recycle orphan files and orphan means files is not on ETCD but only on S3

Reply:
we can clean S3 logs in recycleUnusedSegIndexes for current build. But what about previous build?
I dont' know, maybe check with @xiaocai2333

Reply:
Already checked with @xiaocai2333.
Clear S3 logs before delete index meta is OK.
I will fix it soon.